src/msg/async/EventKqueue.h

```diff
@@ -25,18 +25,33 @@
 
 class KqueueDriver : public EventDriver {
   int kqfd;
-  struct kevent *events;
+  pthread_t mythread;
+  struct kevent *res_events;
   CephContext *cct;
   int size;
 
+  // Keep what we set on the kqfd
+  struct SaveEvent{
+    int fd;
+    int mask;
+  };
+  struct SaveEvent *sav_events;
```

Comment:
because you save all ongoing kqueue events, I think you need to resize in KqueueDriver::resize_events in case of fd is exceed the init number